app/src/store/models/swap.ts

```diff
@@ -81,6 +82,29 @@ export default class Swap {
     return 'Unknown';
   }
 
+  /**
+   * The numeric swap `failureReason` as a user friendly string
+   */
+  get failureLabel() {
+    switch (this.failureReason) {
+      case LOOP.FailureReason.FAILURE_REASON_OFFCHAIN:
+        return 'Off-chain Failure';
+      case LOOP.FailureReason.FAILURE_REASON_TIMEOUT:
+        return 'On-chain Timeout';
+      case LOOP.FailureReason.FAILURE_REASON_SWEEP_TIMEOUT:
+        return 'Sweep Timeout';
+      case LOOP.FailureReason.FAILURE_REASON_INSUFFICIENT_VALUE:
+        return 'Insufficient Value';
+      case LOOP.FailureReason.FAILURE_REASON_TEMPORARY:
+        return 'Temporary Failure';
+      case LOOP.FailureReason.FAILURE_REASON_INCORRECT_AMOUNT:
+        return 'Incorrect Amount';
+      case LOOP.FailureReason.FAILURE_REASON_NONE:
```

Comment:
`FAILURE_REASON_NONE` is the success case. I'd return 'Success' here and 'Failed` only in the default case. Just to be consistent.

Reply:
This is a catch-all case that theoretically should never be hit. In [HistoryRow.tsx#L122](https://github.com/lightninglabs/lightning-terminal/blob/bb40620d8bc4de74bfe23aca212e1aad431083b0/app/src/components/history/HistoryRow.tsx#L122) it checks if the `swap.state` value is Failed. Only in this case will it display the `failureLabel`. The only way this value would be displayed in the UI is if `swap.state === FAILED && swap.failureReason === FAILURE_REASON_NONE`, which should not be possible. If this does occur, then something is terribly wrong, which is why the message says to check the logs. IMO I don't think it is correct to display "Success" when the swap's `state` is Failed.

Reply:
Okay, I see. If you see this as a fall back error message for a case that should never happen, it makes sense :+1: